samples/subsys/usb/hid/sample.yaml

```diff
@@ -0,0 +1,6 @@
+sample:
+  name: USB HID sample
+tests:
+  test_hid:
+    platform_whitelist: quark_se_c1000_devboard 96b_carbon
```

Comment:
just depend on usb_device, no need to whitelist platforms.

Reply:
depends_on: usb_device

Reply:
@nashif I had this depends but it is missing some 2 tests.
16 tests selected, 36678 tests discarded due to filters
against
18 tests selected, 36678 tests discarded due to filters

Reply:
carbon needs to have usb_device set in the board file.